### PR TITLE
Feature/tao 8928/add navigation service

### PR DESCRIPTION
**Pull request by jsconan**
Related to: https://oat-sa.atlassian.net/browse/TAO-8928

Add a navigation service that manages the test map, allowing to filter it.
The review panel has also been reworked, simplifying it and moving the filter handling to the plugin, the component being only responsible to display the content and to emit events when interacting with it.

Once filtered by the navigation service, a copy of the test map is sent to the test runner.
However, the feature is not complete yet, as a change on the navigator service will be required, to accept corrupted jumps tables. The filtering of the test map made by the navigation service introduces some gaps in the jumps table, preventing the navigator to perform properly. Since the first item could not be at position 0, the plugin next-prev must also be updated accordingly.

These missing parts will be addressed in an other PR.

The PR might look huge, the unit tests have been massively changed. However, the actual payload of the PR is in a few files: the review panel and the navigation service, say 5 files.

Impacted unit tests:
- `taoReview/views/js/test/review/plugins/navigation/review-panel/panel/test.html`
- `taoReview/views/js/test/review/plugins/navigation/review-panel/panel-data/test.html`
- `taoReview/views/js/test/review/plugins/navigation/review-panel/plugin/test.html`
- `taoReview/views/js/test/review/services/navigation-data/test.html`

To launch tests:
```
cd tao/views/build
npx grunt connect:dev taoreviewtest
```

### CODE REVIEW

File: views/js/review/plugins/navigation/review-panel/plugin.js [hunk 1]
```diff
@@ -21,16 +21,35 @@
...
+            return true;
+        },
+
+        /**
+         * Filter for incorrect items
+         * @param {mapEntry} item
+         * @returns {Boolean}
+         */
+        incorrect(item) {
+            return !item.informational && (!item.maxScore || item.score !== item.maxScore);
```
Comment (bziondik) [inline views/js/review/plugins/navigation/review-panel/plugin.js]: If the item doesn't have maxScore (for example write text or send file) it will be considered as incorrect?

File: views/js/review/plugins/navigation/review-panel/panel-data.js [hunk 2]
```diff
@@ -43,15 +50,32 @@ define([
...
+            return 'info';
+        }
+        if (item.maxScore) {
+            if (item.score === item.maxScore) {
+                return 'correct';
+            } else {
+                return 'incorrect';
+            }
+        }
+        return 'default';
```
Comment (bziondik) [inline views/js/review/plugins/navigation/review-panel/panel-data.js]: Here type of item without maxScore is 'default' (see another comment)

File: views/js/review/plugins/navigation/review-panel/plugin.js (hunk 1 above)
Comment (jsconan) [reply views/js/review/plugins/navigation/review-panel/plugin.js]: Yes indeed. Without the max score we cannot know if the item is correct or no. So incorrect by default.

File: views/js/review/plugins/navigation/review-panel/panel-data.js (hunk 2 above)
Comment (jsconan) [reply views/js/review/plugins/navigation/review-panel/panel-data.js]: Yes, without max score, we cannot know if it is correct or incorrect, so the status is unknown.
However, this is a edge case, as the max score needs to be set for the test review to properly work

Comment (jsconan) [reply views/js/review/plugins/navigation/review-panel/panel-data.js]: Yes, without max score, we cannot know if it is correct or incorrect, so the status is unknown.
However, this is a edge case, as the max score needs to be set for the test review to properly work
